### PR TITLE
rust: allow more args

### PR DESCRIPTION
**Pull request by saolsen**

### CODE REVIEW

File: languages/rust/oso/src/host/from_polar.rs [hunk 1]
```diff
@@ -213,3 +214,482 @@ where
...
+where
+    A: FromPolar,
+    B: FromPolar,
+    C: FromPolar,
+{
+    fn from_polar(_term: &Term, _host: &mut Host) -> crate::Result<Self> {
+        Err(crate::OsoError::FromPolar)
+    }
+
+    fn from_polar_list(terms: &[Term], host: &mut Host) -> crate::Result<Self> {
```
Comment (dhatch) [inline languages/rust/oso/src/host/from_polar.rs:227]: Remove default impl of this.

File: languages/rust/oso/src/host/from_polar.rs [hunk 2]
```diff
@@ -213,3 +214,482 @@ where
...
     }
 }
+impl<A, B, C> FromPolar for (A, B, C)
+where
+    A: FromPolar,
+    B: FromPolar,
+    C: FromPolar,
+{
+    fn from_polar(_term: &Term, _host: &mut Host) -> crate::Result<Self> {
+        Err(crate::OsoError::FromPolar)
```
Comment (dhatch) [inline languages/rust/oso/src/host/from_polar.rs:224]: Why are these the same trait if the implementation is an error for every term?

File: languages/rust/oso/src/host/method.rs [hunk 3]
```diff
@@ -83,3 +267,345 @@ where
...
+    ) -> Self::Result {
+        (self)(
+            receiver, args.0, args.1, args.2, args.3, args.4, args.5, args.6, args.7, args.8,
+            args.9, args.10, args.11, args.12, args.13, args.14, args.15,
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (dhatch) [inline languages/rust/oso/src/host/method.rs:480]: add #[ignore]

File: languages/rust/oso/src/host/from_polar.rs (hunk 2 above)
Comment (saolsen) [reply languages/rust/oso/src/host/from_polar.rs:224]: Yeah probably don't have to be, could do it like ToPolar and to ToPolarResults where one feeds into the other. Don't really wanna mess with it now though.

File: languages/rust/oso/src/host/from_polar.rs (hunk 1 above)
Comment (dhatch) [reply languages/rust/oso/src/host/from_polar.rs:227]: Skip nvm. too hard.

File: languages/rust/oso/src/host/from_polar.rs (hunk 2 above)
Comment (dhatch) [reply languages/rust/oso/src/host/from_polar.rs:224]: On follow up list.

**Review by dhatch — APPROVED** [review]
love me some 16 arg functions.

File: languages/rust/oso/src/host/method.rs [hunk 4]
```diff
@@ -83,3 +267,346 @@ where
...
+    // This would be great if it was a proc macro but I don't wanna set up another crate.
```
Comment (samscott89) [inline languages/rust/oso/src/host/method.rs:481]: This seems like a decent solution:
https://docs.rs/impl-trait-for-tuples/0.1.3/impl_trait_for_tuples/

Comment (samscott89) [reply languages/rust/oso/src/host/method.rs:481]: Might not work for method though since the trait isn't on the tuple itself, it's the assoc type.

**Comment by saolsen** [comment]
No real time to mess with this today but I'll add it as part of the clean up list.

**Pull request merged**